lib/src/solver/result.dart

```diff
@@ -105,6 +94,35 @@ class SolveResult {
   SolveResult(this._root, this._previousLockFile, this.packages, this.pubspecs,
       this.availableVersions, this.attemptedSolutions, this.resolutionTime);
 
+  /// Checks that the SolveResult is compatible with [_previousLockfile]
+  ///
+  /// Throws if pubspec.yaml isn't satisfied.
+  Future<void> enforceLockfile() async {
+    for (final package in packages) {
+      if (package.isRoot) continue;
+      final previousPackage = _previousLockFile.packages[package.name];
+      if (previousPackage == null) {
+        throw ApplicationException(
```

Comment:
Consider using the summary as error message.
We can make the summary as a list of changes only and omit the `(x.y.z available)` notices (might need to tweak code for this).

Then we can make the error message something like:
```
Resolving...
+ tar 0.5.5+1
- foo 1.2.3
> bar 1.2.3 (was 1.0.0)
Resolution of pubspec.yaml does not satisfy pubspec.lock, which is required by --enforce-lockfile.
```